web/src/features/moreCast2/components/TabbedDataGrid.tsx

```diff
@@ -0,0 +1,406 @@
+            { columnName: 'rh', visible: false },
+            { columnName: 'precip', visible: false },
+            { columnName: 'windDirection', visible: false },
+            { columnName: 'windSpeed', visible: false }
+          ],
+          columnVisibilityModel
+        )
+      )
+    }
+  }, [forecastSummaryVisible]) // eslint-disable-line react-hooks/exhaustive-deps
+
+  /********** End useEffects for managing visibility of column groups *************/
+
+  const [clickedColDef, setClickedColDef] = useState<GridColDef | null>(null)
+
+  // Updates forecast field for a given weather parameter (temp, rh, precip, etc...) based on the
+  // model/source selected in the column header menu
+  const updateColumnWithModel = (modelType: ModelType, colDef: GridColDef) => {
+    // The value of coldDef.field will be precipForecast, rhForecast, tempForecast, etc.
+    // We need the prefix to help us grab the correct weather model field to update (eg. tempHRDPS,
+    // precipGFS, etc.)
+    const forecastField = colDef.field as keyof MoreCast2Row
+    const index = forecastField.indexOf('Forecast')
+    const prefix = forecastField.slice(0, index)
+    const actualField = `${prefix}Actual` as keyof MoreCast2Row
+    modelType === ModelChoice.YESTERDAY
+      ? updateColumnFromLastActual(forecastField, actualField)
+      : updateColumnFromModel(modelType, forecastField, actualField, prefix)
+  }
+
+  // Persistence forecasting. Get the most recent actual and persist it through the rest of the
+  // days in this forecast period.
+  const updateColumnFromLastActual = (forecastField: keyof MoreCast2Row, actualField: keyof MoreCast2Row) => {
+    const newRows = [...visibleRows]
+    // Group our visible rows by station code and work on each group sepearately
+    const groupedByStationCode = groupBy(newRows, 'stationCode')
+    for (const values of Object.values(groupedByStationCode)) {
+      // Get rows with actuals that have non-NaN values
+      const rowsWithActuals: MoreCast2Row[] = values.filter(value => !isNaN(value[actualField] as number))
+      // Filter for the row with the most recent forDate as this contains our most recent actual
+      const mostRecentRow = rowsWithActuals.reduce((a, b) => {
+        return a.forDate > b.forDate ? a : b
+      })
+      // The most recent value from the weather station for the weather parameter of interest
+      // (eg. tempActual) which will be applied as the forecast value
+      const mostRecentValue = mostRecentRow[actualField]
+      // Finally, get an array of rows that don't have an actual for the weather parameter of interest
+      // and iterate through them to apply the most recent actual as the new forecast value
+      const rowsWithoutActuals: MoreCast2Row[] = values.filter(value => isNaN(value[actualField] as number))
+      rowsWithoutActuals.forEach(row => {
+        const predictionItem = row[forecastField] as PredictionItem
+        predictionItem.choice = ModelChoice.YESTERDAY
```

Comment:
might want to consider changing the name of this to `ModelChoice.PERSISTENCE` - if we're using the term "persistence" on the FE & with users, I suspect after a while we'll forget what "Yesterday" even means when we return to this code later

Reply:
Done!